pruner/service.go

```diff
@@ -88,12 +92,19 @@ func (s *Service) Stop(ctx context.Context) error {
 
 	select {
 	case <-s.doneCh:
-		return nil
 	case <-ctx.Done():
 		return fmt.Errorf("pruner unable to exit within context deadline")
 	}
+
+	err := storeCheckpoint(ctx, s.ds, s.checkpoint)
+	if err != nil {
+		return fmt.Errorf("pruner: saving checkpoint on stop: %w", err)
+	}
+
+	return nil
 }
 
+// TODO: Exposed but accesses checkpoint without synchronization
 func (s *Service) LastPruned(ctx context.Context) (uint64, error) {
 	err := s.loadCheckpoint(ctx)
```

Comment:
should we lock checkpoint mux in load checkpoint and where accessing last pruned header? feels like needs read protection no?

Reply:
It is required if LastPruned called after Pruner.Start

Reply:
Is there a reason that this method isn't protected and relies on the callers to protect it though? Like what's the point of leaving it unprotected?

Reply:
I made sure synchronization is everywhere now